File: app/src/main/java/com/example/moviemvp/network/ApiClient.java
package com.example.moviemvp.network;


import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    private static final String BASE_URL = "http://api.themoviedb.org/3/";
    private static Retrofit retrofit = null;
    public static final String IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w185/";
    public static final String BACKDROP_BASE_URL = "https://image.tmdb.org/t/p/w185/";

    public static Retrofit getClient() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/example/moviemvp/network/ApiClient.kt
package com.example.moviemvp.network


import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ApiClient {

    private val BASE_URL = "http://api.themoviedb.org/3/"
    private var retrofit: Retrofit? = null
    const val IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w185/"
    const val BACKDROP_BASE_URL = "https://image.tmdb.org/t/p/w185/"

    val client: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit
        }
}